ci is failing due to clippy error on doctest signature in gh but not locally. root cause and fix.

Clippy is now failing on main when it passed on a previous PR. See [here](https://github.com/momentohq/client-sdk-rust/actions/runs/9101191898/job/25022190033).

I don't know why clippy doesn't fail locally even though the commands are the same. I will address the error directly:


```
error: bound is defined in more than one place
  --> test-util/src/test_utils.rs:20:22
   |
20 | pub fn doctest<'ctx, Fn: 'ctx, Fut: 'ctx>(func: Fn) -> DoctestResult
   |                      ^^
21 | where
22 |     Fn: FnOnce(String, CredentialProvider) -> Fut,
   |     ^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_bound_locations
   = note: `-D clippy::multiple-bound-locations` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::multiple_bound_locations)]`

error: bound is defined in more than one place
  --> test-util/src/test_utils.rs:20:32
   |
20 | pub fn doctest<'ctx, Fn: 'ctx, Fut: 'ctx>(func: Fn) -> DoctestResult
   |                                ^^^
...
23 |     Fut: Future<Output = DoctestResult>,
   |     ^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_bound_locations

error: could not compile `momento-test-util` (lib) due to 2 previous errors
make: *** [Makefile:11: lint] Error 101
```

I will adjust the trait bounds and see if that fixes it on GH.

--- a/test-util/src/test_utils.rs
+++ b/test-util/src/test_utils.rs
@@ -17,10 +17,10 @@ pub type DoctestResult = anyhow::Result<()>;
 /// - Reading the auth token from the environment
 /// - Creating a cache for the doctest to use.
 /// - Ensuring that cache is deleted, even if the test case panics.
-pub fn doctest<'ctx, Fn: 'ctx, Fut: 'ctx>(func: Fn) -> DoctestResult
+pub fn doctest<'ctx, Fn, Fut>(func: Fn) -> DoctestResult
 where
-    Fn: FnOnce(String, CredentialProvider) -> Fut,
-    Fut: Future<Output = DoctestResult>,
+    Fn: 'ctx + FnOnce(String, CredentialProvider) -> Fut,
+    Fut: 'ctx + Future<Output = DoctestResult>,
 {
     let runtime = tokio::runtime::Builder::new_current_thread()
         .enable_all()

